### DISCUSSION TITLE
Feature request: input_select support

### DISCUSSION THREAD

**Discussion by willliamchan** [Clooos/Bubble-Card#301] [Feature requests]
Ever since I start using input_select and I can never find one single card that really cool. Looking forward to see what Clooos will do about it with Bubble Card.

**Comment by Clooos** [comment]
It's finally coming!

https://www.reddit.com/r/BubbleCard/comments/1ds7tay/hi_everyone_today_i_took_some_time_to_work_on_one/

**Reply by willliamchan** [reply]
![image](https://github.com/Clooos/Bubble-Card/assets/54702221/72281777-4b1f-4373-b085-21d2833e2757)

I like it! But instead of having a fixed one on the right, can we do it with a sub-button too?

**Reply by Clooos** [reply]
You are the second one to ask for that, I guess that I will have to add that possibility then 🤓

**Reply by willliamchan** [reply]
> You are the second one to ask for that, I guess that I will have to add that possibility then 🤓

Once and for all, if I can click on the icon or sub button and it gives me the select and that will be perfect.

I can already think of different use case like TV select, Fan speed, swing etc - If this is something doable too I believe we don’t actually need to use “more-info” that much coz everything are in the card itself.

**Reply by Clooos** [reply]
This is going to be a lot harder than what I thought but I'm on it, the next beta will add select support for the sub-buttons 👌

But this will probably be a breaking change for some custom styles (I hope not). I still need to look for the best solution.

**Reply by willliamchan** [reply]
> This is going to be a lot harder than what I thought but I'm on it, the next beta will add select support for the sub-buttons 👌
> 
> But this will probably be a breaking change for some custom styles (I hope not). I still need to look for the best solution.

No rush and take your time!

**Reply by Clooos** [reply]
This is even more challenging than I expected, but I’ve finally made progress!

![image](https://github.com/Clooos/Bubble-Card/assets/36499953/7dd11762-73c9-4ae5-ac82-078db545202d)

**Reply by willliamchan** [reply]
> This is even more challenging than I expected, but I’ve finally made progress!
> 
> ![image](https://private-user-images.githubusercontent.com/36499953/346176103-7dd11762-73c9-4ae5-ac82-078db545202d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxOTcwMzIsIm5iZiI6MTcyMDE5NjczMiwicGF0aCI6Ii8zNjQ5OTk1My8zNDYxNzYxMDMtN2RkMTE3NjItNzNjOS00YWU1LWFjODItMDc4ZGI1NDUyMDJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA1VDE2MjUzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0YTBiMWVkMzE5MTQ0ZWJmZDIyYWZkN2YzZjg2MzA3ODhhMDE0YzJhMjhjNzBiMDRiMWUzZWI4ZjY2ZWM1OTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lkluDTUP-EZqzJlqpYfVyv6-26cWH5t7dcgicvpJwGE)

Look so cool! Wondering if the additional arrow is a must? I am on iPhone and I am afraid we might not be able to display more than 3 sub buttons. Attached an example for your reference.

![IMG_1078](https://github.com/Clooos/Bubble-Card/assets/54702221/eb348b80-3164-4847-8dc4-fd36c0d3f884)

**Reply by Clooos** [reply]
I will add the ability to hide it 👌

**Comment by willliamchan** [comment]
Hey mate! Just a quick bug report, the select somehow can't stays foreground. Please also check out below how I use the new select feature incorporate with area clean option! don't rush to fix it. We all know how busy you and your partner are these days!

![Input_select_not_stays_foreground](https://github.com/Clooos/Bubble-Card/assets/54702221/6ef56e39-2d92-4c02-8e87-f69252a44f3f)

**Reply by Clooos** [reply]
I'm aware of this issue, this will be fixed in the next beta.

**Comment by willliamchan** [comment]
Is the below something that we can put to your To-Do list?

![image](https://github.com/Clooos/Bubble-Card/assets/54702221/7a54221a-120f-45c0-b4e1-beff4151f5b5)
![image](https://github.com/Clooos/Bubble-Card/assets/54702221/eef81a3c-885f-4c1f-bffa-16ac55413d08)

**Reply by Clooos** [reply]
This is already in my to-do list 😁

**Comment by willliamchan** [comment]
Just tried 2.1.0b4 and another bug report for you mate!

![Input_select_not_stays_foreground](https://github.com/user-attachments/assets/7372dae5-028a-45d6-b225-4be6866b948c)

**Reply by Clooos** [reply]
Damn... This was supposed to be fixed 😅

Can you share the YAML of your Favorites section?

**Reply by willliamchan** [reply]
> Damn... This was supposed to be fixed 😅
> 
> Can you share the YAML of your Favorites section?

haha ofcoz! here you go!
[Favotires.txt](https://github.com/user-attachments/files/16202013/Favotires.txt)

**Reply by Clooos** [reply]
Well don't tell me why, probably a CSS limitation but this is because of this custom style on the card with the select sub-button:

```yaml
                ha-card {
                  opacity: 0.95;
                }
```

Without it there is no issue and any other CSS styling on the ha-card element is working with no issue as well.

There is probably nothing that I can do to fix that 😅

**Reply by willliamchan** [reply]
> Well don't tell me why, probably a CSS limitation but this is because of this custom style on the card with the select sub-button:
> 
> ```yaml
>                 ha-card {
>                   opacity: 0.95;
>                 }
> ```
> 
> Without it there is no issue and any other CSS styling on the ha-card element is working with no issue as well.
> 
> There is probably nothing that I can do to fix that 😅

Got it thank you clooos!

**Reply by Clooos** [reply]
You're welcome!
